### PR TITLE
Support for list of primitive enums

### LINKED ISSUE #1125: [Bug] Cannot generate function with a list of primitive enums.
### Describe the bug

Using a `Vec<MyEnum>` as a parameter or return type for a function causes the generated code to have errors.

For example:
```rs
pub fn list_of_enums_in_function(weekdays: Vec<Weekdays>) -> Vec<Weekdays> {
    weekdays
}
```

Error for parameter + generated code:
**bridge_generated.io.rs**

```
no function or associated item named `new_with_null_ptr` found for type `i32` in the current scope
items from traits can only be used if the trait is implemented and in scope

`NewWithNullPtr` defines an item `new_with_null_ptr`, perhaps you need to implement it
```

```rs
#[no_mangle]
pub extern "C" fn new_list_weekdays_0(len: i32) -> *mut wire_list_weekdays {
    let wrap = wire_list_weekdays {
        ptr: support::new_leak_vec_ptr(<i32>::new_with_null_ptr(), len),
        len,
    };
    support::new_leak_box_ptr(wrap)
}
```

Error for return type + generated code:
**bridge_generated.rs**

```
the trait bound `Vec<api::Weekdays>: flutter_rust_bridge::IntoDart` is not satisfied
the following other types implement trait `flutter_rust_bridge::IntoDart`:
  Vec<DateTime<Local>>
  Vec<DateTime<Utc>>
  Vec<NaiveDateTime>
  Vec<T>
  Vec<Uuid>
  Vec<chrono::Duration>
  Vec<f32>
  Vec<f64>
and 8 others

required by a bound in `wrap`
```

```rs
fn wire_list_of_enums_in_function_impl(
    port_: MessagePort,
    weekdays: impl Wire2Api<Vec<Weekdays>> + UnwindSafe,
) {
    FLUTTER_RUST_BRIDGE_HANDLER.wrap(
        WrapInfo {
            debug_name: "list_of_enums_in_function",
            port: Some(port_),
            mode: FfiCallMode::Normal,
        },
        move || {
            let api_weekdays = weekdays.wire2api();
            move |task_callback| Ok(list_of_enums_in_function(api_weekdays))
        },
    )
}
```

Error + generated code:
Dart **bridge_generated.io.dart**

```
The method '_api_fill_to_wire_weekdays' isn't defined for the type 'FlutterRustBridgeExampleSingleBlockTestPlatform'.
Try correcting the name to the name of an existing method, or defining a method named '_api_fill_to_wire_weekdays'.
```

```dart
@protected
  ffi.Pointer<wire_list_weekdays> api2wire_list_weekdays(List<Weekdays> raw) {
    final ans = inner.new_list_weekdays_0(raw.length);
    for (var i = 0; i < raw.length; ++i) {
      _api_fill_to_wire_weekdays(raw[i], ans.ref.ptr[i]);
    }
    return ans;
  }
```

### Codegen logs with `RUST_LOG=debug` environment variable

```shell
No error is generated.
```

### To Reproduce

Add this code to `frb_example/pure_dart/rust/src/api.rs` and run codegen:

```rs
pub fn list_of_enums_in_function(weekdays: Vec<Weekdays>) -> Vec<Weekdays> {
    weekdays
}
```

### Expected behavior

No errors in generated files.

### Generated binding code

_No response_

### OS

_No response_

### Version of `flutter_rust_bridge_codegen`

_No response_

### Flutter info

_No response_

### Version of `clang++`

_No response_

### Version of `ffigen`

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by erikas-taroza**
## Changes
Fixes #1125

This doesn't apply only to functions. For example, structs can now have a list of primitive enums.

## Checklist

- [x] An issue to be fixed by this PR is listed above.
- [x] New tests are added to ensure new features are working. End-to-end tests are usually in the `./frb_example/pure_dart` example, more specifically, `rust/src/api.rs` and `dart/lib/main.dart`.
- [x] The code generator is run and the code is formatted (via `just precommit`).
- [ ] If this PR adds/changes features, documentations (in the `./book` folder) are updated.
- [x] CI is passing.

## Remark for PR creator

- Justfile is a task runner for the command line interface that allows you to run shell commands from a file named `justfile` in your project directory. You can use Justfile after [installing it](https://github.com/casey/just). Note that commands written in `justfile` of this repository are expected to be run in `bash`, not `cmd` or `powershell`. Running `just ...` commands in `cmd` or `powershell` will produce errors as the syntax is not compatible. On Windows, you can use `git bash` if you have Git installed.
- If fzyzcjy does not reply for a few days, maybe he just did not see it, so please ping him.

### CODE REVIEW

**Comment by erikas-taroza** [comment]
@fzyzcjy Can you please rerun [this CI](https://github.com/fzyzcjy/flutter_rust_bridge/actions/runs/4431592331)? The error says it timed out but every other test passed.

**Comment by fzyzcjy** [comment]
Reruned. No worries, as long as the other ios ci thinks ok, it is usually ok :)

Good job! Ping me when ready for review

**Comment by erikas-taroza** [comment]
> Reruned. No worries, as long as the other ios ci thinks ok, it is usually ok :)
> 
> Good job! Ping me when ready for review

Thank you! I was just waiting on the CI. I have to go soon so I will mark it ready for review.

**Comment by fzyzcjy** [comment]
I am quite busy now so will review probably within a day :)

**Comment by erikas-taroza** [comment]
> I am quite busy now so will review probably within a day :)

Sounds good :)

File: frb_codegen/src/generator/dart/ty_general_list.rs [hunk 1]
```diff
@@ -12,15 +12,33 @@ impl TypeDartGeneratorTrait for TypeGeneralListGenerator<'_> {
...
+            format!(
                 "final ans = inner.new_{ident}_{context}(raw.length);
                 for (var i = 0; i < raw.length; ++i) {{
                     _api_fill_to_wire_{inner}(raw[i], ans.ref.ptr[i]);
                 }}
                 return ans;
                 "
-            )),
+            )
+        };
```
Comment (fzyzcjy) [inline frb_codegen/src/generator/dart/ty_general_list.rs:38]: nit: looks like the if vs else branch only differs by the line inside for loop (ans.ref.ptr[i] vs api_fill_to_wire). Thus maybe we can make this code a bit more concise, by extracting out common parts and only write if/else for the different thing

**Review by fzyzcjy — COMMENTED** [review]
Good job! Only one nit and ready to merge

**Comment by erikas-taroza** [comment]
> Good job! Only one nit and ready to merge

Done :)

**Comment by fzyzcjy** [comment]
Great!

**Pull request merged**